api-models/course.yaml

```
@@ -0,0 +1,154 @@
+openapi: 3.0.0
+info:
+  title: Courses Microservice API
+  description: This API manages different types of courses in SkillVerse
+  version: 1.0.0
+
+paths:
+  /courses:
+    get:
+      summary: List all courses
+      operationId: listCourses
```

```suggestion
      operationId: getAllCourses
```
To be consistent with the "get by ID" operation id